src/v/redpanda/application.cc

```diff
@@ -2400,7 +2401,15 @@ void application::start_runtime_services(
           .get();
     }
     syschecks::systemd_message("Starting controller").get();
-    controller->start(cd, app_signal.abort_source()).get0();
+    ss::shared_ptr<cluster::cloud_metadata::offsets_upload_requestor>
+      offsets_upload_requestor;
+    if (offsets_upload_router.local_is_initialized()) {
+        offsets_upload_requestor = offsets_upload_router.local_shared();
+    }
```

Comment:
I imagine that offsets_upload_requestor being null can be fine, right?

Reply:
Good catch -- it shouldn't be, but I think it's possible and should be fixed. Will follow up